certmon/httpd/status.go

```diff
@@ -0,0 +1,41 @@
+package httpd
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/Symantec/Dominator/lib/html"
+	"io"
+	"net/http"
+)
+
+func (s state) statusHandler(w http.ResponseWriter, req *http.Request) {
+	if req.URL.Path != "/" {
+		http.NotFound(w, req)
+		return
+	}
+	writer := bufio.NewWriter(w)
+	defer writer.Flush()
+	fmt.Fprintln(writer, "<title>keymaster-certmond status page</title>")
+	fmt.Fprintln(writer, `<style>
+                          table, th, td {
+                          border-collapse: collapse;
+                          }
+                          </style>`)
+	fmt.Fprintln(writer, "<body>")
+	fmt.Fprintln(writer, "<center>")
+	fmt.Fprintln(writer, "<h1>keymaster-certmond status page</h1>")
+	fmt.Fprintln(writer, "</center>")
+	html.WriteHeaderWithRequest(writer, req)
+	fmt.Fprintln(writer, "<h3>")
+	s.writeDashboard(writer)
+	for _, htmlWriter := range htmlWriters {
+		htmlWriter.WriteHtml(writer)
+	}
+	fmt.Fprintln(writer, "</h3>")
+	fmt.Fprintln(writer, "<hr>")
+	html.WriteFooter(writer)
+	fmt.Fprintln(writer, "</body>")
+}
```

Comment:
This should definetively be made into an html template... but that is for another dat

Reply:
I've been thinking of making a library function/package for this. I think that's a better fit than a template. There are several *Dominator* servers that could use this.